### PR TITLE
sphere and distance field stuff.

### PR DESCRIPTION
**Pull request by acornacorn**

### CODE REVIEW

File: collision_detection_distance_field/include/moveit/collision_detection_distance_field/collision_common_distance_field.h [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+#ifndef MOVEIT_COLLISION_DETECTION_DISTANCE_FIELD_COLLISION_COMMON_DISTANCE_FIELD_
+#define MOVEIT_COLLISION_DETECTION_DISTANCE_FIELD_COLLISION_COMMON_DISTANCE_FIEL
+
+#include <moveit/collision_detection/world.h>
+#include <moveit/collision_detection/collision_world.h>
+
+namespace collision_detection
+{
+
+extern const AllowedCollisionMatrix empty_acm;
```
Comment (isucan) [inline collision_detection_distance_field/include/moveit/collision_detection_distance_field/collision_common_distance_field.h:46]: Do we need this in many locations? Could we not define hidden, local (static) instances and where this is needed?

File: collision_detection_distance_field/include/moveit/collision_detection_distance_field/collision_robot_distance_field.h [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+
+  // sphere centers for all spheres for all links in link order
+  EigenSTL::vector_Vector3d sphere_centers_;
+
+  // sphere radii for all spheres for all links in link order
+  std::vector<double> sphere_radii_;
+
+  // for each sphere, index of link the sphere is associated with.
+  // This is an index into the kmodel_->getLinkModels() and state->getLinkStateVector() vectors
+  std::vector<uint16_t> sphere_link_map_;
```
Comment (isucan) [inline collision_detection_distance_field/include/moveit/collision_detection_distance_field/collision_robot_distance_field.h:195]: 65535 seems perhaps large, but i can imagine people wanting to do horribly crazy things; am i missing some optimization later on that makes use of this?

File: collision_detection_distance_field/src/collision_robot_distance_field.cpp [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+  res_ = res;
+  state1_ = state1;
+  state2_ = state2;
+  other_robot_ = other_robot;
+  other_state1_ = other_state1;
+  other_state2_ = other_state2;
+  acm_ = acm;
+
+  // debug
+  if (1)
```
Comment (isucan) [inline collision_detection_distance_field/src/collision_robot_distance_field.cpp:76]: maybe add a verbose_ flag?

File: collision_detection_distance_field/include/moveit/collision_detection_distance_field/collision_common_distance_field.h (hunk 1 above)
Comment (acornacorn) [inline collision_detection_distance_field/include/moveit/collision_detection_distance_field/collision_common_distance_field.h:46]: I'm going to eliminate this soon.

File: collision_detection_distance_field/include/moveit/collision_detection_distance_field/collision_robot_distance_field.h (hunk 2 above)
Comment (acornacorn) [inline collision_detection_distance_field/include/moveit/collision_detection_distance_field/collision_robot_distance_field.h:195]: Good point.  I'll add a typedef so it is easy to change if people need more spheres.

File: collision_detection_distance_field/src/collision_robot_distance_field.cpp (hunk 3 above)
Comment (acornacorn) [inline collision_detection_distance_field/src/collision_robot_distance_field.cpp:76]: I may hook this up to the verbose flag later.  I was just going to delete it later when I don't need it.

**Pull request merged**